### PR TITLE
Update benchmark branch with latest develop

### PR DESCRIPTION
**Pull request by erikh360**
Most of the commit are from the develop branch so it doesn't need a full review, the two relevant ones are:
https://github.com/praekelt/junebug/pull/168/commits/cdc0e66af4d827e2c0c5eaf89d282681ff1e9cc2
https://github.com/praekelt/junebug/pull/168/commits/14c1112bd1854a3aa40a813a8604144ba27b1f56

### CODE REVIEW

File: bench/run_benchmarks.py [hunk 1]
```diff
@@ -66,13 +66,7 @@ def create_channel(self):
                     'type': 'dmark',
                     'config': {
                         'web_path': 'api',
-                        'web_port': 8001,
-                        "redis_manager": {
-                          "key_prefix": "my_ussd",
-                          "host": "redis",
-                          "db": 2,
-                          "port": 6379
-                        }
```
Comment (rudigiesler) [inline bench/run_benchmarks.py]: Why are we getting rid of this redis configuration?

File: bench/run_benchmarks.py [hunk 2]
```diff
@@ -112,7 +100,7 @@ def _wait_for_channel_start(self):
 
     def delete_ussd_channel(self, channelid):
         self.conn.request(
-            "DELETE", '/channels/%s' % channelid)
+            "DELETE", '/channels/%s' % channelid, json.dumps({}))
```
Comment (rudigiesler) [inline bench/run_benchmarks.py]: I'm a little unsure why we need a body here? Since it's a delete request.

File: bench/run_benchmarks.py (hunk 1 above)
Comment (erikh360) [reply bench/run_benchmarks.py]: I added it when I was running redis, rabbit and Junebug using docker-compose. I ended up running `jb` directly which meant the default `localhost` was fine.

File: bench/run_benchmarks.py (hunk 2 above)
Comment (erikh360) [reply bench/run_benchmarks.py]: The delete was failing and started working when I added this. Doesn't really make sense. Let me try without it and see.

**Comment by rudigiesler** [comment]
Benchmarks to show influence of routing (type=USSD, concurrency=10, num_messages=1000):
Normal messaging:
```
Average throuhput: 63msgs/s
Average latency: 156.4ms 95% latency: 195.3ms, 99% latency: 212.6ms
```
Routing messaging:
```
Average throuhput: 49msgs/s
Average latency: 200.2ms 95% latency: 249.2ms, 99% latency: 302.9ms
```
About a 22% reduction in throughput.

**Comment by rudigiesler** [comment]
Running benchmarks on pypy (type=USSD, concurrency=10, num_messages=2000, warmup=1000)
Normal messaging:
```
Average throuhput: 74msgs/s
Average latency: 131.9ms 95% latency: 202.1ms, 99% latency: 281.7ms
```
Routing messaging:
```
Average throuhput: 54msgs/s
Average latency: 181.2ms 95% latency: 243.3ms, 99% latency: 454.4ms
```
Routing messaging with 100 destinations:
```
Average throuhput: 45msgs/s
Average latency: 217.2ms 95% latency: 348.0ms, 99% latency: 401.1ms
```
So about a 27% reduction in throughput when using routing, and 39% throughput reduction when having 100 destinations configured.

**Review by rudigiesler — APPROVED** [review]

**Pull request merged**